backup-restore/uninstall/uninstall-backup-restore.sh

```diff
@@ -0,0 +1,238 @@
+# Make sure you are logged into the correct cluster.
+# Execute on a Linux with Bash
+
+USAGE="Usage: $0 [-u] [-s] [-b <Bakup and Restore Name Space>]
+       -u to unistall a spoke installation before uninstalling hub 
+       -s to skip creation of DeleteBackupRequest"
+
+NAMESPACE=ibm-backup-restore
+FORCE=false
+SKIP=false
+
+err_exit()
+{
+        echo "ERROR: $@" >&2
+        exit 1
+}
+
+#SYS_OS=$(uname -s)
+#[ "$SYS_OS" != "Linux" ] && err_exit "$0 should be executed on a Linux system"
+
+
+while getopts "sub:" OPT
+do
+  # echo "ARGS = $OPT : $OPTARG"
+  case ${OPT} in
+    b )
+      NAMESPACE="${OPTARG}"
+      ;;
+    s )
+      SKIP=true
+      ;;
+    u )
+      FORCE=true
+      ;;
+    \? )
+      err_exit "$USAGE"
+      ;;
+    : )
+      err_exit "$USAGE"
+      ;;
+  esac
+done
+
+check_cmd ()
+{
+   (type $1 > /dev/null) || err_exit "$1  command not found"
+}
+
+check_cmd oc
+check_cmd jq
+check_cmd envsubst
+
+oc whoami > /dev/null || err_exit "Not logged in a cluster"
+
+
+print_heading()
+{
+  echo
+  echo -e "===================================================================================================="
+  echo -e "$@"
+  echo -e "===================================================================================================="
+}
+
+
+# Track time to set future expectations
+UNINSTALL_STARTED="$(date +%s)"
+SPOKE=false
+
+ISF_NS=$(oc get spectrumfusion -A --no-headers | cut -d" " -f1)
+[ -z "$ISF_NS" ] &&  ISF_NS=$NAMESPACE
+export ISF_NS
+
+CONNECTION=$(oc -n $NAMESPACE get cm guardian-configmap -o custom-columns="CONN:data.connectionName" --no-headers)
+if [ -n "$CONNECTION" ]
+  then
+     CONN_STAT=$(oc -n ${ISF_NS} get connection $CONNECTION -o custom-columns=S:status.connectionFromRemoteClusterHealth.status --no-headers)
+     SPOKE=true
+     HUB_STATUS=$(oc -n $NAMESPACE get dataprotectionagent -o custom-columns=H:status.hubStatus --no-headers)
+     if [ "$HUB_STATUS" == "Found" ] && [ "$FORCE" != "true" ]
+      then
+         err_exit 'Hub exist, uninstall hub first or use -u option to force uninstall spoke'
+     fi
+fi
+
+if [ "$SKIP" != true ]
+ then
+    print_heading "Remove any existing backups"
+    BACKUPS=$(oc get -n ${ISF_NS} backups.data-protection.isf.ibm.com | grep isf-backup-restore | awk '{print $1}' 2> /dev/null)
+    for BACKUP in ${BACKUPS[@]}; do
+        export BACKUP
+
+        YAML=$(cat <<EOF
+apiVersion: data-protection.isf.ibm.com/v1alpha1
+kind: DeleteBackupRequest
+metadata:
+  annotations:
+    dp.isf.ibm.com/provider-name: isf-backup-restore
+  name: delete-backup-$BACKUP
+  namespace: $ISF_NS
+spec:
+  backup: $BACKUP
+EOF
+        )
+        echo "$YAML" | oc apply -f -
+    done
+
+    UNFINISHED=$(oc -n $ISF_NS get  deletebackuprequest.data-protection.isf.ibm.com -o custom-columns=NAME:metadata.name,STATUS:status.phase --no-headers | egrep -iv "DeleteBackupRequestFailed|Cancelled|Completed|FailedValidation" )
+
+    while [ -n "$UNFINISHED" ]
```

Comment:
is there a small possibility of an endless loop?

Reply:
In theory, yes. And that is by design to user will notice that some deletebackuprequest are not finishing. User can investigate why and fix so that backup get deleted.